test/e2e/daemon_pod_test.go

```diff
@@ -0,0 +1,63 @@
+// +build functional
+
+package e2e
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/argoproj/argo/v3/pkg/apis/workflow/v1alpha1"
+	"github.com/argoproj/argo/v3/test/e2e/fixtures"
+)
+
+type DaemonPodSuite struct {
+	fixtures.E2ESuite
+}
+
+func (s *DaemonPodSuite) TestWorkflowCompletesIfContainsDaemonPod() {
+	hasSucceeded := func(wf *v1alpha1.Workflow) bool {
+		return wf.Status.Phase == v1alpha1.WorkflowSucceeded
+	}
+	s.Given().
+		Workflow(`apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: whalesay-
+  labels:
+    argo-e2e: true
+spec:
+  entrypoint: whalesay
+  templates:
+  - name: whalesay
+    dag:
+      tasks:
+        - name: redis
+          template: redis-tmpl
+        - name: whale
+          dependencies: [redis]
+          template: whale-tmpl
+  - name: redis-tmpl
+    daemon: true
+    container:
+      image: argoproj/argosay:v2
+      args: ["sleep", "100s"]
+  - name: whale-tmpl
+    container:
+      image: argoproj/argosay:v2
+      args: ["echo", "hello world"]
+`).
+		When().
+		SubmitWorkflow().
+		WaitForWorkflow(fixtures.Condition(hasSucceeded), "to show succeeded").
+		Then().
+		ExpectWorkflow(func(t *testing.T, metadata *v1.ObjectMeta, status *v1alpha1.WorkflowStatus) {
+			assert.False(t, status.FinishedAt.IsZero())
```

Comment:
Redundant?

Reply:
Not redundant... this was the issue in question